internal/service/server.go

```diff
@@ -0,0 +1,48 @@
+package service
+
+import (
+	"fmt"
+	"net"
+
+	pb "github.com/brexhq/substation/proto"
+	"google.golang.org/grpc"
+)
+
+// New returns a configured gRPC server.
+func New(opt ...grpc.ServerOption) *grpc.Server {
+	return grpc.NewServer(opt...)
+}
+
+// Server wraps a gRPC server and provides methods for managing server state.
+type Server struct {
+	server *grpc.Server
+}
+
+// Setup creates a new gRPC server.
+func (s *Server) New(opt ...grpc.ServerOption) {
```

Comment:
New isn't adding much value here so it should be removed. It's worth mentioning that we don't do this consistently throughout the project. Here are some examples:
- internal/aws packages use this pattern:
  - https://github.com/brexhq/substation/blob/main/internal/aws/dynamodb/dynamodb.go#L15-L51
  - https://github.com/brexhq/substation/blob/main/internal/aws/lambda/lambda.go#L15-L51
  - https://github.com/brexhq/substation/blob/main/internal/aws/secretsmanager/secretsmanager.go#L15-L51
- internal/http does not use this pattern:
  - https://github.com/brexhq/substation/blob/main/internal/http/http.go#L24-L32

I think the pattern that's broken here is ... 
- Not setting baseline configs in the New function
- Not using a method called `Setup` for initialization 

The example of `internal/http` is actually bad because today I would refactor that to return a new HTTP client that has X-Ray enabled (similar to [this](https://github.com/brexhq/substation/blob/main/internal/sink/http.go#L59-L61)). 

I think the fix here is getting rid of the New function (we can bring it back later if needed) and renaming the method. What do you think?